### ISSUE TITLE
auto_join_rooms not working =(

### ISSUE THREAD

**Issue by psyapathy** [matrix-org/synapse#2732]
Hi all!

In homeserver.yaml:
```
auto_join_rooms:
    - "#all:matrix_server"
```
but when users connect to the server they do not come automatically to this room :(

**Comment by turt2live** [comment]
Possible duplicate of https://github.com/matrix-org/synapse/issues/2725

Did you use the built-in script or Riot to register the user? Existing users won't be affected.

**Comment by Half-Shot** [comment]
Is this still an issue, or can it be closed?

**Comment by krombel** [comment]
I would close it and let it reopen it once people see this issue again as #2725 got closed in the meantime

**Comment by dykstranet** [comment]
I managed to make auto_join_rooms working only after I made the room public, and chose "Who can access this room?" to be "Anyone who knows the room link, including guests". I think the documentation needs to be updated to tell the public room requirement.
